src/fixes/extra_blank_line.rs

```diff
@@ -0,0 +1,188 @@
+use super::Fix;
+use crate::common::*;
+
+pub(crate) struct ExtraBlankLineFixer<'a> {
+    name: &'a str,
+}
+
+impl Default for ExtraBlankLineFixer<'_> {
+    fn default() -> Self {
+        Self {
+            name: "ExtraBlankLine",
+        }
+    }
+}
+
+impl Fix for ExtraBlankLineFixer<'_> {
+    fn name(&self) -> &str {
+        self.name
+    }
+
+    fn fix_warnings(
+        &self,
+        warnings: Vec<&mut Warning>,
+        lines: &mut Vec<LineEntry>,
+    ) -> Option<usize> {
+        let mut count: usize = 0;
+
+        // check and remove all blank lines.
+        let mut is_preview_line_blank = false;
+        lines.clone().iter().enumerate().for_each(|(i, line)| {
```

Comment:
I think we should not clone the entire collection.

You can use `dedup_by` function from Rust std library (https://doc.rust-lang.org/std/vec/struct.Vec.html#method.dedup_by).

Something like
```
lines.dedup_by(|a, b| a.is_empty() && b.is_empty());
```

Reply:
I didn't know that function, thank you.